### PR TITLE
Edit renovate section

### LINKED ISSUE #7433: Add renovate section in package.json
Renovate has been trying hard to pin dependencies for a while now.
We can ask it to not make any attempts for pinning dependencies, by doing:

📌 Important: Renovate will wait until you have merged this Pin PR before creating any upgrade PRs for the affected packages. Add the preset :preserveSemverRanges your config if you instead don't wish to pin dependencies.

Add this preset `preserveSemverRanges` in the config

How the config is structured: https://renovatebot.com/docs/presets-config/

### PR DESCRIPTION
**Pull request by pradeepgangwar**
Fixes #7433 

Here's the Pull Request Doc
https://firefox-dev.tools/debugger.html/docs/pull-requests.html

### Summary of Changes

* Edit renovate config to stop making frequent updates to dependencies.

### CODE REVIEW

**Comment by pradeepgangwar** [comment]
@AnshulMalik Please review.

File: renovate.json [hunk 1]
```diff
@@ -1,7 +1,17 @@
 {
+  "separateMajorMinor": false,
   "extends": [
-    "config:base"
+    ":preserveSemverRanges",
+    "group:all",
```
Comment (AnshulMalik) [inline renovate.json]: I think we can just add `:preserveSemverRanges` and `config:base` for the purpose

And monthly schedule sounds nice. But I think it'll try to accommodate all changes in one PR and it'd be hard to figure out which dependency is breaking the build(if it fails)

And IMO the lockFileMaintenance block is redundant.

Please let me know what you think about these points.

Comment (pradeepgangwar) [reply renovate.json]: I think yes monthly updates will be difficult to handle if build is failing. But if we do not pass that option then what will the frequency of updates?

Comment (AnshulMalik) [reply renovate.json]: They'll come as soon as a new version of a dependency is published.

Comment (pradeepgangwar) [reply renovate.json]: Then I guess I should remove monthly scheduling for update.

**Comment by pradeepgangwar** [comment]
@AnshulMalik Did the changes. Please review.

**Comment by jasonLaster** [comment]
Thanks @pradeepgangwar

**Pull request merged**